test/pg.ddl.js

```diff
@@ -0,0 +1,293 @@
+);
+
+ALTER TABLE "Schema1" ADD CONSTRAINT "pkSchema1Id" PRIMARY KEY ("Id");
+
+-- Category: Schema2 -----------------------------------------------------------
+
+CREATE TABLE "Schema2" (
+  "Id"     bigserial,
+  "master" bigint NOT NULL
+);
+
+ALTER TABLE "Schema2" ADD CONSTRAINT "pkSchema2Id" PRIMARY KEY ("Id");
+
+ALTER TABLE "Schema2" ADD CONSTRAINT "fkSchema2master" FOREIGN KEY ("master") \
+REFERENCES "Schema1" ("Id") ON UPDATE RESTRICT ON DELETE CASCADE;`,
+});
+
+test({
+  name: 'generateDDL with Many',
+  schemas: {
+    Schema1: {
+      field: { domain: 'Nomen' },
+    },
+    Schema2: {
+      field: { domain: 'Nomen' },
+      link: many('Schema1'),
+    },
+  },
+  expectedSql: `${defaultDomainsSql}
+-- Category: Schema1 -----------------------------------------------------------
+
+CREATE TABLE "Schema1" (
+  "Id"    bigserial,
+  "field" text
+);
+
+ALTER TABLE "Schema1" ADD CONSTRAINT "pkSchema1Id" PRIMARY KEY ("Id");
+
+-- Category: Schema2 -----------------------------------------------------------
+
+CREATE TABLE "Schema2" (
+  "Id"    bigserial,
+  "field" text
+);
+
+ALTER TABLE "Schema2" ADD CONSTRAINT "pkSchema2Id" PRIMARY KEY ("Id");
+
+
+CREATE TABLE "Schema2link" (
+  "Schema2" bigint NOT NULL,
+  "Schema1" bigint NOT NULL
+);
+ALTER TABLE "Schema2link" ADD CONSTRAINT "fkSchema2linkSchema2" \
+FOREIGN KEY ("Schema2") REFERENCES "Schema1" ("Id") \
+ON UPDATE RESTRICT ON DELETE CASCADE;
+ALTER TABLE "Schema2link" ADD CONSTRAINT "fkSchema2linkSchema1" \
+FOREIGN KEY ("Schema1") REFERENCES "Schema2" ("Id") \
+ON UPDATE RESTRICT ON DELETE CASCADE;
+ALTER TABLE "Schema2link" ADD CONSTRAINT "pklinkSchema2Schema1" \
+PRIMARY KEY ("Schema2", "Schema1");`,
+});
+
+test({
+  name: 'generateDDL with foreign key before category',
+  schemas: {
+    Schema1: {
+      field: { domain: 'Nomen' },
+      schema2: { category: 'Schema2' },
+    },
+    Schema2: {
+      field: { domain: 'Nomen' },
+    },
+  },
+  expectedSql: `${defaultDomainsSql}
+-- Category: Schema1 -----------------------------------------------------------
+
+CREATE TABLE "Schema1" (
+  "Id"      bigserial,
+  "field"   text,
+  "schema2" bigint
+);
+
+ALTER TABLE "Schema1" ADD CONSTRAINT "pkSchema1Id" PRIMARY KEY ("Id");
+
+-- Category: Schema2 -----------------------------------------------------------
+
+CREATE TABLE "Schema2" (
+  "Id"    bigserial,
+  "field" text
+);
+
+ALTER TABLE "Schema2" ADD CONSTRAINT "pkSchema2Id" PRIMARY KEY ("Id");
+
+ALTER TABLE "Schema1" ADD CONSTRAINT "fkSchema1schema2" \
+FOREIGN KEY ("schema2") REFERENCES "Schema2" ("Id") \
+ON UPDATE RESTRICT ON DELETE SET NULL;`,
+});
+
+test({
+  name: 'generateDDL with all domaint types',
+  schemas: {
+    domains: {
+      Text: { type: 'string' },
+      BigInt: { type: 'number' },
+      Floating: { type: 'number', floating: true },
+      Date: { type: 'object', class: 'Date' },
+      Bytes: { type: 'object', class: 'Uint8Array' },
+      Function: { type: 'function' },
+    },
```

Comment:
You should also test all [predefined domains](https://github.com/metarhia/globalstorage/blob/master/lib/pg.utils.js#L33).